### PR TITLE
kubeswitch: init at 0.7.2

### PR DESCRIPTION
**Pull request by bryanasdev000**
###### Description of changes

kubeswitch: init at 0.7.2

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 192009` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubeswitch</li>
  </ul>
</details>

File: pkgs/development/tools/kubeswitch/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "-s" "-w"
+    "-X github.com/danielfoehrkn/kubeswitch/cmd/switcher.version=${version}"
+    "-X github.com/danielfoehrkn/kubeswitch/cmd/switcher.buildDate=1970-01-01"
+
+  ];
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/switch version | grep ${version} > /dev/null
+  '';
```
Comment (sikmir) [inline pkgs/development/tools/kubeswitch/default.nix:28]:
```suggestion
  passthru.tests.version = testers.testVersion {
    package = kubeswitch;
  };
```

File: pkgs/development/tools/kubeswitch/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  postInstall = ''
+    mv $out/bin/main $out/bin/switch
+  '';
+
+  meta = with lib; {
+    description = "The kubectx for operators";
+    license = licenses.asl20;
+    homepage = "https://github.com/danielfoehrKn/kubeswitch";
+    maintainers = with maintainers; [ bryanasdev000 ];
```
Comment (sikmir) [inline pkgs/development/tools/kubeswitch/default.nix:38]:
```suggestion
    maintainers = with maintainers; [ bryanasdev000 ];
    mainProgram = "switch";
```

File: pkgs/development/tools/kubeswitch/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{ lib, buildGoModule, fetchFromGitHub }:
```
Comment (sikmir) [inline pkgs/development/tools/kubeswitch/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, testers, kubeswitch }:
```

File: pkgs/development/tools/kubeswitch/default.nix (hunk 1 above)
Comment (bryanasdev000) [reply pkgs/development/tools/kubeswitch/default.nix]: Doh, I always forget this :P 

Thanks!

File: pkgs/development/tools/kubeswitch/default.nix (hunk 2 above)
Comment (bryanasdev000) [reply pkgs/development/tools/kubeswitch/default.nix:37]: TIL mainProgram.

**Review by sikmir — APPROVED** [review]

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 192009` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubeswitch</li>
  </ul>
</details>

**Pull request merged**